packages/iov-ethereum/src/ethereumconnection.spec.ts

```diff
@@ -269,43 +268,127 @@ describe("EthereumConnection", () => {
       const blockHeader = await connection.getBlockHeader(0);
       expect(blockHeader.id).toMatch(/^0x[0-9a-f]{64}$/);
       expect(blockHeader.height).toEqual(0);
       expect(blockHeader.transactionCount).toBeGreaterThanOrEqual(0);
-      connection.disconnect();
     });
 
     it("throws error from invalid block number", async () => {
       pendingWithoutEthereum();
-      const connection = await EthereumConnection.establish(base);
+      const connection = await EthereumConnection.establish(base, undefined);
       await connection
         .getBlockHeader(99999999999999)
         .then(() => fail("promise must be rejected"))
         .catch(err => expect(err).toMatch(/Header 99999999999999 doesn't exist yet/));
-      connection.disconnect();
+    });
+  });
+
+  describe("feed", () => {
+    async function postTransaction(amount: string, memo: string): Promise<PostTxResponse> {
+      const wallet = Secp256k1HdWallet.fromMnemonic(
+        "oxygen fall sure lava energy veteran enroll frown question detail include maximum",
+      );
+      const mainIdentity = await wallet.createIdentity(HdPaths.bip44(60, 0, 0, 1));
+
+      const recipientAddress = "0xE137f5264b6B528244E1643a2D570b37660B7F14" as Address;
+
+      const sendTx: SendTransaction = {
+        kind: "bcp/send",
+        chainId: nodeChainId,
+        signer: mainIdentity.pubkey,
+        recipient: recipientAddress,
+        amount: {
+          quantity: amount,
+          fractionalDigits: 18,
+          tokenTicker: "ETH" as TokenTicker,
+        },
+        gasPrice: {
+          quantity: gasPrice,
+          fractionalDigits: 18,
+          tokenTicker: "ETH" as TokenTicker,
+        },
+        gasLimit: {
+          quantity: gasLimit,
+          fractionalDigits: 18,
+          tokenTicker: "ETH" as TokenTicker,
+        },
+        memo: memo,
+      };
+      const connection = await EthereumConnection.establish(base, ws);
+      const senderAddress = ethereumCodec.keyToAddress(mainIdentity.pubkey);
+      const nonceResp = await connection.getNonce({ address: senderAddress });
+      const signingJob = ethereumCodec.bytesToSign(sendTx, nonceResp.data[0]);
+      const signature = await wallet.createTransactionSignature(
+        mainIdentity,
+        signingJob.bytes,
+        signingJob.prehashType,
+        nodeChainId,
+      );
+
+      const signedTransaction: SignedTransaction = {
+        transaction: sendTx,
+        primarySignature: {
+          nonce: nonceResp.data[0],
+          pubkey: mainIdentity.pubkey,
+          signature: signature,
+        },
+        otherSignatures: [],
+      };
+      const bytesToPost = ethereumCodec.bytesToPost(signedTransaction);
+
+      const resultPost = await connection.postTx(bytesToPost);
+      return resultPost;
+    }
+
+    describe("watchBlockHeaders", () => {
+      it("watches headers with same data as getBlockHeader", async done => {
+        pendingWithoutEthereum();
```

Comment:
Using `async done` together causes all kind of strangeness in Jasmine, e.g. pending() does not work anymore. Thus we should make this either `async` or use th `done` callback.

In this case I recommend a `async`+sleep implementation like I did in https://github.com/iov-one/iov-core/commit/1e9f80d845c24239f4de031cd2d1ac0dd52745ad